system/jlib/jmetrics.hpp

```diff
@@ -262,4 +273,50 @@ std::shared_ptr<CustomMetric<T>> createCustomMetricAndAddToReporter(const char *
     return pMetric;
 }
 
+
+class jlib_decl ScopedGaugeUpdater
+{
+public:
+    explicit ScopedGaugeUpdater(const std::shared_ptr<GaugeMetric> &_pGauge, int64_t _amount=1) :
+        gauge{*_pGauge},
```

Comment:
I doesn't seem to any good reason to keep this taking a shared_ptr type, instead of the underlying type itself.

As it stands, someone can't pass in a GaugeMetric type, forcing the usage to unecessarily use shared_ptr type.
Imagine a sequence of code that wants to take a metric and pass it around and then somewhere deeper use an updater, it doesn't necessarily want to pass around a reference to a shared_ptr. It can if it wants, it depends on the use case, but the updater doesn't need it.

You could define it as:
```
    explicit ScopedGaugeUpdater(GaugeMetric *_pGauge, int64_t _amount=1) :
        gauge{*_pGauge},
```
and that would be more correct/flexible than having it take a reference to a shared_ptr,
but shared_ptr doesn't define an overload operator to pointer, so you would have to call .get() on the shared_ptr, e.g.:
```
    hpccMetrics::ScopedGaugeUpdater updater(myGauge.get());
```
which is fine, but you could (and I think I would) make it take a reference to the underlying type, i.e.:
```
    explicit ScopedGaugeUpdater(GaugeMetric &_pGauge, int64_t _amount=1) :
        gauge{_pGauge},
```
And since shared_ptr does overload the dereference operator, you can then use as, e.g:
```
    hpccMetrics::ScopedGaugeUpdater updater(*myGauge);
```

NB: we tend to use Owned in our code base, and passing in an Owned would automatically invoke the operator and pass in the underlying ptr to the object.